Shell/main.cpp

```diff
@@ -906,7 +907,21 @@ static int run_command(const String& cmd)
                 int rc = execvp(argv[0], const_cast<char* const*>(argv.data()));
                 if (rc < 0) {
                     if (errno == ENOENT) {
-                        fprintf(stderr, "%s: Command not found.\n", argv[0]);
+                        int shebang_fd = open(argv[0], O_RDONLY);
+                        auto close_argv = ScopeGuard([shebang_fd]() { if (shebang_fd >= 0)  close(shebang_fd); });
+                        char shebang[256] {};
+                        ssize_t num_read = -1;
+                        if ((shebang_fd >= 0) && ((num_read = read(shebang_fd, shebang, sizeof(shebang))) >= 2) && (StringView(shebang).starts_with("#!"))) {
+                            ssize_t newline_pos = 2;
+                            for (; newline_pos < num_read; ++newline_pos) {
```

Comment:
Should I just add `size_t AK::StringView::find_first_of(const StringView& chars)` instead? this works...

Reply:
Yes, that would be a nice and expressive API. Let's return an `Optional<size_t>` though, in case we don't find anything :)

Reply:
What, comparing against `AK::StringView::NO_MATCH` doesn't work for you? 😉. Sounds good, i'll push a slightly nicer looking fix.